Modules/_testcapi/long.c

```diff
@@ -141,13 +141,155 @@ pylong_aspid(PyObject *module, PyObject *arg)
 }
 
 
+static PyObject *
+layout_to_dict(const PyLongLayout *layout)
```

Comment:
BTW, you can use `Py_BuildValue("{skskslsl}", ...)` for this. It is much more compact.